vim-patch:8.0.1394,8.1.1967,8.2.{23,1000,1059,2354,2419,2433,2473,2850}

janlazo opened the pull request:
https://github.com/neovim/neovim/pull/14532/commits/629984a583cee4ed6b7904f758c4f81ff4539d10 doesn't fail in ASAN so the vim patch likely fixed the `tabonly` crash.

janlazo committed:
vim-patch:8.1.1967: line() only works for the current window

Problem:    Line() only works for the current window.
Solution:   Add an optional argument for the window to use.
https://github.com/vim/vim/commit/8e0a8e7eb7c177807f44db6b76d8e52314248ab5

janlazo committed:
fixup! refactor nvim_buf_set_lines to use extmark_splice

janlazo committed:
vim-patch:8.2.0023: command line editing not sufficiently tested

Problem:    Command line editing not sufficiently tested.
Solution:   Add more tests. (Dominique Pelle, closes vim/vim#5374)
https://github.com/vim/vim/commit/59cb041d0a56d8555857da7e063ec61504ee1fa7

Ported to commit test_ex_mode.vim for later patches.
Test fails because feedkeys() filters out modifier keys on output.

janlazo committed:
vim-patch:8.2.1000: get error when leaving Ex mode with :visual

Problem:    Get error when leaving Ex mode with :visual and a CmdLineEnter
            autocommand was used.
Solution:   Reset ex_pressedreturn. (closes vim/vim#6293)
https://github.com/vim/vim/commit/158ea175a99fc23eae1b0a5ee9a81cdd973854a6

janlazo committed:
vim-patch:8.2.1059: crash when using :tabonly in an autocommand

Problem:    Crash when using :tabonly in an autocommand. (Yegappan Lakshmanan)
Solution:   Do not allow the autocommand window to be closed.
https://github.com/vim/vim/commit/cf8441704d6e517bda1899f4afa82c6b4eecbaec

E813 error message does not mention 'popup' window
because Neovim floating window are regular windows, unlike Vim popups.

https://github.com/neovim/neovim/pull/14532#discussion_r631731829

janlazo committed:
vim-patch:8.2.2354: crash with a weird combination of autocommands

Problem:    Crash with a weird combination of autocommands.
Solution:   Increment b_nwindows when needed. (closes vim/vim#7674)
https://github.com/vim/vim/commit/797e63b9f2baa1853e7063aac478d663cd02f207

janlazo committed:
vim-patch:8.2.2433: opening cmdline window gives error in BufLeave autocommand

Problem:    Opening cmdline window gives error in BufLeave autocommand.
Solution:   Reset cmdwin_type when triggering the autocommand.
https://github.com/vim/vim/commit/b63f3ca66db20f7b135aae706d36be538f91f8cc

janlazo committed:
vim-patch:8.2.2473: crash when leaving command line window triggers autocommand

Problem:    Crash when leaving command line window triggers autocommand.
            (houyunsong)
Solution:   Make sure not to close the current window or buffer.
https://github.com/vim/vim/commit/8c6951fa2836a1ae3257770e7b927a9380439912

N/A patches for version.c:

vim-patch:8.2.2414: using freed memory when closing the cmdline window

Problem:    Using freed memory when closing the cmdline window.
Solution:   Check the window is still valid.
https://github.com/vim/vim/commit/b7e2670b6a1de02c772af5097ba24f2a15b26eec

janlazo committed:
lint

janlazo committed:
Revert "XXX: ex_tabonly(): aucmd_win is not part of the window list."

This reverts commit 7214d0bc846179a862e8d3061d00270a6caa0d7b.

Cannot test patch v8.2.1059 without reverting this commit.

janlazo committed:
vim-patch:8.2.2359: strange test failure with MS-Windows

Problem:    Strange test failure with MS-Windows.
Solution:   Skip the system() call for now.
https://github.com/vim/vim/commit/e2924328c165f1fc549f91bf212c93c87bb1d9ed

janlazo committed:
vim-patch:8.2.2419: autocmd test was failing on MS-Windows with GUI

Problem:    Autocmd test was failing on MS-Windows with GUI.
Solution:   Remove stray feedkeys().
https://github.com/vim/vim/commit/d697ddea1466d877a79f8c56d46361dc54e9baef

bfredl commented on src/nvim/globals.h:
```diff
@@ -987,6 +987,8 @@ EXTERN char_u e_dirnotf[] INIT(= N_(
 EXTERN char_u e_au_recursive[] INIT(= N_(
     "E952: Autocommand caused recursive behavior"));
+EXTERN char_u e_autocmd_close[] INIT(= N_(
+    "E813: Cannot close autocmd or popup window"));
```
only `autocmd window` I think?

janlazo replied:
Do we not use this for `floating` windows or do we not make a distinction between floating and regular windows for autocmds?

I don't mind removing `or popup` but I won't update the translations.

bfredl replied:
AFAIK we only use this for autocmd window. Floats can always be closed just like regular windows.

bfredl commented on src/nvim/window.c:
```diff
@@ -6394,6 +6394,10 @@ void restore_win_noblock(win_T *save_curwin, tabpage_T *save_curtab,
   if (win_valid(save_curwin)) {
     curwin = save_curwin;
     curbuf = curwin->w_buffer;
+  } else if (curwin->w_floating) {
```
I am not sure this should be done. floating windows can be entered as ordinary windows in neovim, unlike vim popups.

janlazo replied:
I'll consider it N/A then. Up to you to port if needed.

janlazo committed:
vim-patch:8.0.1394: cannot intercept a yank command

Problem:    Cannot intercept a yank command.
Solution:   Add the TextYankPost autocommand event. (Philippe Vaucher et al.,
            closes vim/vim#2333)
https://github.com/vim/vim/commit/7e1652c63c96585b9e2235c195a3c322b1f11595

janlazo committed:
fixup! vim-patch:8.0.1449: slow redrawing with DirectX

janlazo committed:
fixup! vim-patch:8.1.1228: not possible to process tags with a function

janlazo committed:
vim-patch:8.2.2850: recalling commands from history is not tested

Problem:    Recalling commands from history is not tested.
Solution:   Add tests. (closes vim/vim#8194)
https://github.com/vim/vim/commit/71c6f7a665eb467054fbf319989d97c252af28d2
